### PR TITLE
Fixed issue - #19346 Import data 2.2.6 Value for 'product_type' attribute contains incorrect value

### PR DESCRIPTION
**Pull request by GovindaSharma**
### Description (*)
Fixed issue - #19346
Import data 2.2.6 Value for 'product_type' attribute contains incorrect value

Since 'product_attribute' is reserved attribute,so we can restrict this type of attribute to be created by user.

### Fixed Issues (if relevant)

Fixed issue - #19346
Import data 2.2.6 Value for 'product_type' attribute contains incorrect value

### Manual testing scenarios (*)

   1. create an attribute with code product_type http://i.prntscr.com/AwUJuhizQKatBwNwRa1S7Q.png with type: dropdown scope: global sortable, comparable: no
  2.  add it to a family and create a product.
   3. go to import profiles download sample file then reimport sample file

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19408) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @GovindaSharma. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php [hunk 1]
```diff
@@ -115,6 +115,14 @@ public function execute()
             $response->setError(true);
             $response->setProductAttribute($attribute->toArray());
         }
+        if(in_array($attributeCode, \Magento\Catalog\Model\ResourceModel\Eav\Attribute::DEFAULT_ATTRIBUTE)){
+    		$message = 	__('Cannot create attribute (%1) as it is already used in catalog',$attributeCode);
+    		$this->setMessageToResponse($response, [$message]);
+    		$response->setError(true);
+    		$response->setProductAttribute($attribute->toArray());
```
Comment (orlangur) [inline app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: Please add a proper check into

```
if ($attribute->getId() && !$attributeId) {
            $message = strlen($this->getRequest()->getParam('attribute_code'))
                ? __('An attribute with this code already exists.')
                : __('An attribute with the same code (%1) already exists.', $attributeCode);
```

condition, do not introduce a constant - just compare with `'product_type'` string literal.

**Review by orlangur — CHANGES_REQUESTED** [review]
Please fill all description fields according to template and apply code changes, then squash it into single commit and force push.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php (hunk 1 above)
Comment (GovindaSharma) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: so instead of introducing a constant...i should compare the string..if string is similar then throw msg right?

**Comment by GovindaSharma** [comment]
changes done as per request

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: @GovindaSharma exactly, no need to overcomplicate things. When we need more than one such forbidden not-really-an-attribute, we may think about array.

**Comment by GovindaSharma** [comment]
@orlangur can you please tell me what is the exact issue in first check which is not passed?i have tested the code in php 7.2 version and it's working.Please let me the exact issue so that i can fix the same

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: @GovindaSharma,

```
if(strcmp($attributeCode,'product_type')==false){
    		$message = 	__('Cannot create attribute (%1) as it is already used in catalog',$attributeCode);
```

still not combined with first condition as requested. `$attributeCode !== 'product_type'` should do the trick.

**Comment by orlangur** [comment]
@GovindaSharma,

> can you please tell me what is the exact issue in first check which is not passed?

I don't understand this question, please reword.

**Comment by GovindaSharma** [comment]
since two checks has been passed but one is not..so what issue occurred in that checkes for which it has not been passed?

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php (hunk 1 above)
Comment (GovindaSharma) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: hi @orlangur 

"Please fill all description fields according to template and apply code changes, then squash it into single commit and force push."

this was your feedback..i did not understand it.can you please tell me give any idea what changes i need to apply in code?

**Comment by GovindaSharma** [comment]
@orlangur please look into the changes and let me know what other things i need to do?

**Comment by GovindaSharma** [comment]
@magento-engcom-team and @orlangur ,can you please review the changes as i have done all necessary changes suggested by @orlangur.Let me know your feedback on this.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: @GovindaSharma check this: https://github.com/magento/magento2/pull/19408/files#r236223352

You need to add new check into existing `if` instead of introducing a new one.

File: app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php [hunk 2]
```diff
@@ -37,7 +37,6 @@ class Attribute extends \Magento\Eav\Model\Entity\Attribute implements
     const ENTITY = 'catalog_eav_attribute';
 
     const KEY_IS_GLOBAL = 'is_global';
-
     /**
```
Comment (orlangur) [inline app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php]: Revert this change)

Comment (GovindaSharma) [reply app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php]: are you talking about the extra space in file? that i need to remove?

**Comment by GovindaSharma** [comment]
@magento-engcom-team and @orlangur .i have done changes requested by @orlangur .Please review this and let me know.

File: app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php (hunk 2 above)
Comment (GovindaSharma) [reply app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php]: @orlangur .i made the chnages.Please review this.
Thank You.

File: app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php (hunk 1 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Controller/Adminhtml/Product/Attribute/Validate.php]: @GovindaSharma still not fixed

File: app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php (hunk 2 above)
Comment (orlangur) [reply app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php]: Try `git checkout 2.3-develop -- app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php` to get this file in correct state.

Comment (GovindaSharma) [reply app/code/Magento/Catalog/Model/ResourceModel/Eav/Attribute.php]: @orlangur file is in correct state now.I made the changes.

**Comment by orlangur** [comment]
@GovindaSharma please check https://github.com/magento/magento2/pull/19408/files#r237182111. All needed changes are not done yet.

**Comment by GovindaSharma** [comment]
@orlangur , now i have added check into existing if,previously i added extra if just to show different message because "product_type" attribute in real does not exist in magento,so that was the reason to add extra if.But now i added the checks into existing if condition.Please review the changes and let me know about it.

Thank You

**Comment by orlangur** [comment]
@GovindaSharma thanks! Looks good now, please squash changes into a single commit (incuding my commits) and we are good to go.

**Comment by GovindaSharma** [comment]
@orlangur .I have now squash all commits into one and done force push.I think all the changes has been done now.Please check and approve the changes.
Thank You

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3597](https://jira.corp.magento.com/browse/ENGCOM-3597) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@GovindaSharma thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-engcom-team** [comment]
Hi @GovindaSharma. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Comment by GovindaSharma** [comment]
@magento-engcom-team ,thanks for your confirmation.
